agent/ui_endpoint_test.go

```diff
@@ -2125,6 +2149,42 @@ func TestUIServiceTopology(t *testing.T) {
 				FilteredByACLs: false,
 			},
 		},
+		{
+			name: "cbackend",
+			httpReq: func() *http.Request {
+				req, _ := http.NewRequest("GET", "/v1/internal/ui/service-topology/cbackend?kind=", nil)
+				return req
+			}(),
+			want: &ServiceTopology{
+				Protocol:         "http",
+				TransparentProxy: false,
+				Upstreams:        []*ServiceTopologySummary{},
+				Downstreams: []*ServiceTopologySummary{
+					{
+						ServiceSummary: ServiceSummary{
+							Name:           "cproxy",
+							Datacenter:     "dc1",
+							Tags:           []string{"http", "https"},
```

Comment:
@kisunji  I found a possible bug here. `http` and it's service `cproxy-http` is not connect enabled. Should this tag show up here then or not? Only `cproxy-https` with the tag `https` is connect enabled.

Reply:
And I guess `Nodes` below could get deduped as well like it is done for the tags:
https://github.com/hashicorp/consul/blob/449e0507417d01a4b44f57b64890c4d8dfccfc4a/agent/ui_endpoint.go#L597 ?

Reply:
I pushed a commit to dedupe the nodes in this branch as well: https://github.com/hashicorp/consul/pull/13023/commits/bf3865f819156c1bd3cd14d8073b9a63f7b925c9

Reply:
Regarding my first comment in this thread: For transparent proxies it is checked if all instances are in the same "state" ( https://github.com/hashicorp/consul/blob/449e0507417d01a4b44f57b64890c4d8dfccfc4a/agent/ui_endpoint.go#L586-L588 ). I guess it would make sense to apply a similar code connect-native services -- I will add code towards that end in another commit. If we think this should not be the case we revert / remove it easily again.

Reply:
Actually I am leaning towards a service to be considered connect native if any of it's instances are. Connect native is not exactly the same as transparent proxies and allows for even more flexibility (like taking to only a subset of instances of a service which are connect enabled etc…). Will push a commit towards that goal.

Reply:
Yep, I agree with your decision. Thanks for being thorough with the changes!